Dev/ahn nath: styling and adding the backend to the project

ahn-nath opened the pull request:
- Styled sections and improved the UI of some pages.
- Create the UI and base template for the contributions page. 
- Added the backend.

ahn-nath committed:
feat: create React routes

Created React routes and base pages for each relevant view corresponding to a wireframe.

ahn-nath committed:
fix: deleting node_modules from folder

ahn-nath committed:
feat: connected the amplify backend to project

We have connected the server folder or Express.js based backend of the ProgramEquity team to our project and tested that everything is functioning correctly. We cleaned/removed some files we won't use.

ahn-nath committed:
minor fixes to backend

We are commenting some lines.

ahn-nath committed:
feat: Create UI for the available open source projects feed

Added base component to list each element.

ahn-nath committed:
style: improving IU style

ahn-nath committed:
Merge branch 'main' into dev/ahn-nath

tushar5526 commented on frontend/src/pages/login.js:
```diff
@@ -1,177 +1,14 @@
-import React, { useState, useEffect, useContext } from "react";
```
Why are we deleting the OAuth functionality?

ahn-nath replied:
Because we only need to use a personal token for the GitHub API. At least that is the easiest option,

tushar5526 commented:
Did you pull the changes from the main, switched to a new branch and then worked further?

ahn-nath commented:
> Did you pull the changes from the main, switched to a new branch and then worked further?

Yes, but after I made the changes to my branch. I initially forgot to merge it before continuing on my branch and had to resolve conflicts. Let me know if I deleted something relevant.

tushar5526 commented:
Yeah sorry I forgot that we dont need OAuth. Its good to go!
